dev/Scroller/TestUI/ScrollerDynamicPage.xaml.cs

```diff
@@ -2008,7 +2008,94 @@ private void BtnAddRepeatedSnapPoint_Click(object sender, RoutedEventArgs e)
+        {
+            try
+            {
+                double? horizontalOffset = string.IsNullOrWhiteSpace(txtCOAHO.Text) ? (double?)null : Convert.ToDouble(txtCOAHO.Text);
+                double? verticalOffset = string.IsNullOrWhiteSpace(txtCOAVO.Text) ? (double?)null : Convert.ToDouble(txtCOAVO.Text);
+                float? zoomFactor = string.IsNullOrWhiteSpace(txtCZFAZF.Text) ? (float?)null : Convert.ToSingle(txtCZFAZF.Text);
+                bool disableAnimation = true;
+
+                if (horizontalOffset == null && verticalOffset == null && zoomFactor == null)
+                {
+                    return;
+                }
+
+                ComboBox cmbAnimationMode = (zoomFactor == null || zoomFactor == scroller.ZoomFactor) ? 
+                    cmbScrollAnimationMode : cmbZoomAnimationMode;
+
+                switch (cmbAnimationMode.SelectedIndex)
+                {
+                    case 0: // Disabled
+                        break;
+                    case 1: // Enabled
+                        disableAnimation = false;
+                        break;
+                    default: // Auto
+                        disableAnimation = !new UISettings().AnimationsEnabled;
+                        break;
+                }
+
+                ChangeView(scroller, horizontalOffset, verticalOffset, zoomFactor, disableAnimation);
+            }
+            catch (Exception ex)
+            {
+                txtExceptionReport.Text = ex.ToString();
+                lstScrollerEvents.Items.Add(ex.ToString());
+            }
+        }
+
+        private bool ChangeView(Scroller scroller, double? horizontalOffset, double? verticalOffset, float? zoomFactor, bool disableAnimation)
+        {
+            AppendAsyncEventMessage($"ChangeView(horizontalOffset:{horizontalOffset}, verticalOffset:{verticalOffset}, zoomFactor:{zoomFactor}, disableAnimation:{disableAnimation}) from horizontalOffset:{scroller.HorizontalOffset}, verticalOffset:{scroller.VerticalOffset}, zoomFactor:{scroller.ZoomFactor}");
+
+            double targetHorizontalOffset = horizontalOffset == null ? scroller.HorizontalOffset : (double)horizontalOffset;
+            double targetVerticalOffset = verticalOffset == null ? scroller.VerticalOffset : (double)verticalOffset;
+            float targetZoomFactor = zoomFactor == null ? scroller.ZoomFactor : (float)Math.Max(Math.Min((double)zoomFactor, scroller.MaxZoomFactor), scroller.MinZoomFactor);
+            float deltaZoomFactor = targetZoomFactor - scroller.ZoomFactor;
+
+            if (disableAnimation)
+            {
+                targetHorizontalOffset = Math.Max(Math.Min(targetHorizontalOffset, scroller.ExtentWidth * targetZoomFactor - scroller.ViewportWidth), 0.0);
+                targetVerticalOffset = Math.Max(Math.Min(targetVerticalOffset, scroller.ExtentHeight * targetZoomFactor - scroller.ViewportHeight), 0.0);
+            }
+
+            if (deltaZoomFactor == 0.0f)
```

Comment:
what if both offset and zoom is provided ? does zoomTo make it go to the correct offset ?

Reply:
Yes, thanks to the centerPoint computation, the final offsets will also be correct.

At this point, my opinion is that a ChangeView method should not be added to the Scroller. This method is simple and as you can see it's a matter of calling ScrollTo or ZoomTo based on whether the zoom factor changes or not. By the way, the old ChangeView signature would not be appropriate because somehow the Id of the operation needs to be returned - so what would a ChangeView method return?  ScrollTo returns ScrollInfo and ZoomTo returns ZoomInfo. 
The centerPoint computation is simple:
    Vector2 centerPoint = new Vector2(
        (float)(targetHorizontalOffset * scroller.ZoomFactor - scroller.HorizontalOffset * targetZoomFactor) / deltaZoomFactor,
        (float)(targetVerticalOffset * scroller.ZoomFactor - scroller.VerticalOffset * targetZoomFactor) / deltaZoomFactor);
I frankly don't think it is worth bloating the Scroller API for this.